init.cpp

```diff
@@ -992,22 +978,96 @@ void main_title_menu()
         }
         if (key == u8"f"s)
         {
             snd(20);
             await(400);
-            finish_elona();
-            return;
+            return main_menu_wrapper(main_menu_result_t::finish_elona);
+        }
+    }
+}
+
+main_menu_result_t main_menu_wrapper(main_menu_result_t initial)
+{
+    main_menu_result_t result = initial;
+    bool finished = false;
+    bool game_initialized = false;
+    while(!finished)
+    {
+        switch(result)
+        {
+            // Main menu
+
+        case main_menu_result_t::main_menu_incarnate:
+            result = main_menu_incarnate();
+            break;
+        case main_menu_result_t::main_menu_continue:
+            result = main_menu_continue();
+            break;
+        case main_menu_result_t::main_menu_new_game:
+            result = main_menu_new_game();
+            break;
+        case main_menu_result_t::main_title_menu:
+            // Loop back to the start.
+            result = main_menu_result_t::main_title_menu;
+            finished = true;
+            break;
+
+            // Character making
+
+        case main_menu_result_t::character_making_select_race:
+            result = character_making_select_race();
+            break;
+        case main_menu_result_t::character_making_select_sex:
+            result = character_making_select_sex();
+            break;
+        case main_menu_result_t::character_making_select_sex_looped:
+            result = character_making_select_sex(false);
+            break;
+        case main_menu_result_t::character_making_select_class:
+            result = character_making_select_class();
+            break;
+        case main_menu_result_t::character_making_select_class_looped:
+            result = character_making_select_class(false);
+            break;
+        case main_menu_result_t::character_making_role_attributes:
+            result = character_making_role_attributes();
+            break;
+        case main_menu_result_t::character_making_role_attributes_looped:
+            result = character_making_role_attributes(false);
+            break;
+        case main_menu_result_t::character_making_select_feats_and_alias:
+            result = character_making_select_feats_and_alias();
+            break;
+        case main_menu_result_t::character_making_select_feats_and_alias_looped:
+            result = character_making_select_feats_and_alias(false);
+            break;
+        case main_menu_result_t::character_making_final_phase:
+            result = character_making_final_phase();
+            break;
+
+            // Finished initializing, start the game.
+
+        case main_menu_result_t::initialize_game:
+            result = main_menu_result_t::initialize_game;
+            finished = true;
+            break;
+        case main_menu_result_t::finish_elona:
+            result = main_menu_result_t::finish_elona;
+            finished = true;
```

Comment:
If you select `Exit` in main menu in debug build, app crashes by `assert(0)`. The cause is unintended fall through in this case clause.

Reply:
Whoops, added.